### PR TITLE
gflags: fix build on case-insensitive filesystems

### PR DESCRIPTION
**Pull request by aperiodic**
###### Motivation for this change

Darwin machines come with a case-insensitive filesystem by default. The
gflags package's source contains a file 'BUILD' and the build process
attempts to create a directory called 'build', which fails on
case-insensitive filesystems.

###### Things done

Add a prePatch hook to rename the BUILD file (which is for use with an unrelated build tool).

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions

- [ ] I did not test compilation of all pkgs that depend on this change because this should not produce a change to any machine code
- [x] Tested execution of the bash completion script in bin
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by LnL7** [comment]
@GrahamcOfBorg build gflags

File: pkgs/development/libraries/gflags/default.nix [hunk 1]
```diff
@@ -7,4 +7,9 @@ stdenv.mkDerivation
         sha256 = "03lxc2ah8i392kh1naq99iip34k4fpv22kwflyx3byd2ssycs9xf";
       };
     nativeBuildInputs = [ cmake ];
+    # The unpacked source contains a file called 'BUILD', and cmake creates
+    # a directory called 'build'. To fix building on the case-insensitive
+    # filesystems commonly seen on darwin machines, rename 'BUILD' so it won't
+    # conflict with the directory (it's for the unrelated build tool "bazel").
+    prePatch = "mv BUILD BUILD.bazel";
   }
```
Comment (LnL7) [inline pkgs/development/libraries/gflags/default.nix]: Could you also add meta.platforms, otherwise this won't get built by hydra.

**Comment by aperiodic** [comment]
@LnL7 Added platforms along with some other metadata.

**Comment by LnL7** [comment]
looks like you forgot a semicolon after one of the attributes.

**Comment by LnL7** [comment]
@GrahamcOfBorg build gflags

**Review by LnL7 — APPROVED** [review]
Thanks!

**Comment by LnL7** [comment]
Also ported to release in 021ee7d93be.

**Pull request merged**